src/plugin_discovery/plugin_pack/pack_at_path.js

```diff
@@ -0,0 +1,42 @@
+import { Observable } from 'rxjs';
+import { isAbsolute, resolve } from 'path';
+import { createInvalidPackError } from '../errors';
+
+import { isDirectory, isFile } from './lib';
+import { PluginPack } from './plugin_pack';
+
+async function createPackAtPath(path) {
+  if (!isAbsolute(path)) {
+    throw createInvalidPackError(null, 'requires an absolute path');
+  }
+
+  if (!await isDirectory(path)) {
+    throw createInvalidPackError(path, 'must be a directory');
+  }
+
+  const pkgPath = resolve(path, 'package.json');
+  if (!await isFile(pkgPath)) {
+    throw createInvalidPackError(path, 'must have a package.json file');
+  }
+
+  const pkg = require(pkgPath);
+  if (!pkg || typeof pkg !== 'object') {
+    throw createInvalidPackError(path, 'must have a value package.json file');
+  }
+
+  let provider = require(path);
+  if (provider.__esModule) {
+    provider = provider.default;
+  }
+  if (typeof provider !== 'function') {
+    throw createInvalidPackError(path, 'must export a function');
+  }
+
+  return new PluginPack({ path, pkg, provider });
+}
+
+export const createPackAtPath$ = (path) => (
+  Observable.fromPromise(createPackAtPath(path))
+    .map(pack => ({ pack }))
```

Comment:
maybe a nit? possibly my ignorance. if we don't need the `()` around `{ pack }` can we remove them? it might be clearer

Reply:
They are needed. When you use arrow functions object expressions are a special case, because JavaScript uses curly braces to mark the function’s body in an arrow function. Hopefully the two examples below explain how it works:

```js
.map(pack => { pack })
// is the same as
.map(pack => {
  pack
});
```

while

```js
.map(pack => ({ pack }))
// is the same as
.map(pack => {
  return { pack }
});
```